core/src/main/java/com/netflix/conductor/core/events/ActionProcessor.java

```diff
@@ -88,20 +89,32 @@ public ActionProcessor(WorkflowExecutor executor, MetadataService metadataServic
 
         Map<String, Object> input = new HashMap<>();
         input.put("workflowId", taskDetails.getWorkflowId());
+        input.put("taskId", taskDetails.getTaskId());
         input.put("taskRefName", taskDetails.getTaskRefName());
         input.putAll(taskDetails.getOutput());
 
         Map<String, Object> replaced = parametersUtils.replace(input, payload);
-        String workflowId = "" + replaced.get("workflowId");
-        String taskRefName = "" + replaced.get("taskRefName");
-        Workflow found = executor.getWorkflow(workflowId, true);
+        String workflowId = (String) replaced.get("workflowId");
+        String taskId = (String) replaced.get("taskId");
+        String taskRefName = (String) replaced.get("taskRefName");
+
+        Workflow found = null;
+        if (StringUtils.isNotEmpty(workflowId)) {
+            found = executor.getWorkflow(workflowId, true);
```

Comment:
Potential optimization: when taskId is not, we can get task by taskId directly without loading workflow with all tasks.